tiatoolbox/tools/registration/wsi_registration.py

```diff
@@ -1233,3 +1284,218 @@ def apply_bspline_transform(
 
     sitk_registered_image_sitk = resampler.Execute(moving_image_sitk)
     return sitk.GetArrayFromImage(sitk_registered_image_sitk)
+
+
+class AffineWSITransformer:
+    """Resampling regions from a whole slide image.
+
+    This class is used to resample tiles/patches from a whole slide image
+    using transformation.
+
+    Example:
+        >>> from tiatoolbox.tools.registration.wsi_registration import (
+        ... AffineWSITransformer
+        ... )
+        >>> from tiatoolbox.wsicore.wsireader import WSIReader
+        >>> wsi_reader = WSIReader.open(input_img=sample_ome_tiff)
+        >>> transform_level0 = np.eye(3)
+        >>> tfm = AffineWSITransformer(wsi_reader, transform_level0)
+        >>> output = tfm.read_rect(location, size, resolution=resolution, units="level")
+
+    """
+
+    def __init__(self, wsi_reader: WSIReader, transform: np.ndarray) -> None:
```

Comment:
Should this implement the WSIReader interface and be in wsireader? @shaneahmed

Reply:
Let's keep it here for now. When you start working with Zarr we can move it and implement as WSIReader interface. What do you think?